doc/ARS/Local_ARS_HLD.md

```diff
@@ -0,0 +1,892 @@
+                description "ARS profile Name";
+                type leafref {
+                    path "/sars:sonic-ars/sars:ARS_PROFILE/sars:ARS_PROFILE_LIST/sars:profile_name";
+                }
+            }
+
+            leaf assign_mode {
+                type enumeration {
+                    enum per_flowlet_quality{
+                        description "Per flow-let assignment based on flow quality";
+                    }
+                    enum per_packet {
+                        description "Per packet flow assignment based on port load";
+                    }
+                }
+            }
+
+            leaf flowlet_idle_time {
+                type uint16 {
+                    range 2..2047;
+                }
+                description  "Idle duration in microseconds. This duration is to classifying a flow-let in a macro flow.";
+            }
+
+            leaf max_flows {
+                type uint32;
+                description  "Maximum number of flow states that can be maintained per ARS object.";
+            }
+
+            leaf primary_path_threshold {
+                type uint32;
+                description  "Primary path metric";
+            }
+
+            leaf alternative_path_cost {
+                type uint32;
+                description  "Alternative path cost";
+            }
+
+            leaf-list alternative_path_members {
+                type leafref {
+                    path "/port:sonic-port/port:PORT/port:PORT_LIST/port:name";
+                }
+                description "Members of the LAG participating in alternative path";
+            }
+        }
+        /* end of list ARS_PORTCHANNEL_LIST */
+    }
+    /* end of container ARS_PORTCHANNEL */
+
+```
+
+##### ACL_RULE changes
+
+```
+    container sonic-acl {
+
+        container ACL_RULE {
+
+            description "ACL_RULE part of config_db.json";
+
+            list ACL_RULE_LIST {
+
+                key "ACL_TABLE_NAME RULE_NAME";
+
+                leaf ACL_TABLE_NAME {
+                    type leafref {
+                        path "/acl:sonic-acl/acl:ACL_TABLE/acl:ACL_TABLE_LIST/acl:ACL_TABLE_NAME";
+                    }
+                }
+
+                leaf RULE_NAME {
+                    type string {
+                        length 1..255;
+                    }
+                } 
+...
+                leaf DISABLE_ARS_FORWARDING {
+                    description "Disable ARS forwarding on matching packets";
+                    type boolean;
+                    default false;
+                }
+            }
+         }
+    }
+```
+
+#### Config DB Enhancements  
+
+```
+; New container ARS_PROFILE_TABLE
+; ARS global configuration
+
+key                     = ARS_PROFILE|profile_name
+
+;field                  = value
+
+algorithm               = "ewma"        ;Path quality calculation algorithm
+max_flows               = uint32        ;Maximum number of flows that can be maintained for ARS

```

Comment:
Not the only algo used by vendors. Probably will require other vendors to introduce changes in SAI API